go/integration/hcvault/hcvault_aead_internal_test.go

```diff
@@ -17,75 +17,102 @@
 )
 
-func TestExtractKey(t *testing.T) {
+func TestGetPaths(t *testing.T) {
 	for _, tc := range []struct {
 		desc string
 		uri  string
-		key  string
+		fmt  string
 		err  string
 	}{
 		{
 			desc: "simple",
-			uri:  "hcvault://vault.example.com/hi",
-			key:  "hi",
+			uri:  "hcvault://vault.example.com/transit/keys/foo",
+			fmt:  "transit/%s/foo",
 		},
 		{
-			desc: "path",
-			uri:  "hcvault://vault.example.com/foo/bar/baz",
-			key:  "foo/bar/baz",
+			desc: "escaped",
+			uri:  "hcvault://vault.example.com/transit/keys/this%2Band+that",
+			fmt:  "transit/%s/this%%2Band+that",
 		},
 		{
-			desc: "hyphen host",
-			uri:  "hcvault://vault-prd.example.com/coyote",
-			key:  "coyote",
+			desc: "sub-path",
+			uri:  "hcvault://vault.example.com/teams/billing/something/transit/keys/pci-key",
+			fmt:  "teams/billing/something/transit/%s/pci-key",
 		},
 		{
-			desc: "empty string",
-			uri:  "hcvault://example.com/",
-			key:  "",
+			desc: "transit-twice",
+			uri:  "hcvault://vault.example.com/transit/keys/something/transit/keys/my-key",
+			fmt:  "transit/keys/something/transit/%s/my-key",
 		},
 		{
-			desc: "escaped",
-			uri:  "hcvault://vault.example.com/this%2Band+that",
-			key:  "this%2Band+that",
+			desc: "hyphen-host",
+			uri:  "hcvault://vault-prd.example.com/transit/keys/hi",
+			fmt:  "transit/%v/hi",
+		},
+		{
+			desc: "no-host",
+			uri:  "hcvault:///transit/keys/hi",
+			fmt:  "transit/%v/hi",
 		},
 		{
-			desc: "no host",
-			uri:  "hcvault:///hi",
-			key:  "hi",
+			desc: "mount-not-named-transit",
+			uri:  "hcvault:///cipher/keys/hi",
+			fmt:  "cipher/%v/hi",
 		},
 		{
 			desc: "http",
 			uri:  "http://vault.com/hi",
 			err:  "malformed keyURL",
 		},
 		{
-			desc: "no path",
+			desc: "no-path",
 			uri:  "hcvault://vault.com",
 			err:  "malformed keyURL",
 		},
 		{
-			desc: "slash only",
+			desc: "slash-only",
 			uri:  "hcvault://vault.com/",
-			key:  "",
+			err:  "malformed keyURL",
+		},
+		{
+			desc: "not-transit",
+			uri:  "hcvault://vault.example.com/foo/bar/baz",
+			err:  "malformed keyURL",
+		},
+		{
+			desc: "not-end-of-path",
+			uri:  "hcvault://vault.example.com/transit/keys/bar/baz",
+			err:  "malformed keyURL",
 		},
 	} {
 		t.Run(tc.desc, func(t *testing.T) {
-			key, err := extractKey(tc.uri)
+			encPath, decPath, err := getEndpointPaths(tc.uri)
 			if err == nil {
 				if tc.err != "" {
 					t.Fatalf("Missing error, want=%s", tc.err)
 				}
+				exp := fmt.Sprintf(tc.fmt, encryptSegment)
+				if encPath != exp {
+					t.Fatalf("Incorrect encrypt path, want=%s;got=%s", exp, encPath)
+				}
+				exp = fmt.Sprintf(tc.fmt, decryptSegment)
+				if decPath != exp {
+					t.Fatalf("Incorrect decrypt path, want=%s;got=%s", exp, decPath)
+				}
 			} else {
 				if tc.err != err.Error() {
 					t.Fatalf("Incorrect error, want=%s;got=%s", tc.err, err)
 				}
-			}
-			if key != tc.key {
-				t.Fatalf("Incorrect key, want=%s;got=%s", tc.key, key)
+				if encPath != "" {
```

Comment:
Changed to `t.Errorf` in 65ddde686. Left in the checks. I personally have been burned when a value was unexpected set, to I like to make sure they're not, and figure it's harmless enough, yes?

Reply:
Sounds good.

One more thing here... since the zero values of `tc.enc` and `tc.dec` are empty strings, you can move checking them out of the `err` conditional. Something like:

```
encPath, decPath, err := getEndpointPaths(tc.uri)
if err == nil {
	if tc.err != "" {
		t.Errorf("getEndpointPaths(%q) err is nil, want %q", tc.uri, tc.err)
	}
} else {
	if tc.err != err.Error() {
		t.Errorf("getEndpointPaths(%q) err = %v; want %q", tc.uri, err, tc.err)
	}
}

if encPath != tc.enc {
	t.Errorf("getEndpointPaths(%q) encryptPath = %q, want %q", tc.uri, encPath, tc.enc)
}
if decPath != tc.dec {
	t.Errorf("getEndpointPaths(%q) decryptPath = %q, want %q", tc.uri, decPath, tc.dec)
}
```

Note that I'm also suggesting a change to the error messages to better align with our best practices (see https://google.github.io/styleguide/go/decisions#identify-the-function).

Reply:
Done in a3f60e6a3.